lib/mocha/mock.rb

```diff
@@ -109,14 +109,15 @@ class Mock
     #   object.expects(:expected_method_one).returns(:result_one)
     #   object.expects(:expected_method_two).returns(:result_two)
     def expects(method_name_or_hash, backtrace = nil)
-      iterator = ArgumentIterator.new(method_name_or_hash)
-      iterator.each do |*args|
+      CompositeExpectation.new(Array(method_name_or_hash).map do |*args|
```

Comment:
Looking at the documentation:

`@return [Expectation] last-built expectation which can be further modified by methods on {Expectation}`

This introduces a breaking change. So, even if we agree that the new behavior is preferable, we'd have to think about deprecating the old behavior.

There are no existing tests to check for the documented behavior, though.